NLSY/nlsy_impute.R

```diff
@@ -0,0 +1,160 @@
+test$race_w <- ifelse(test$race == "White", 1, 0)
+test$race_b <- ifelse(test$race == "Black or African American", 1, 0)
+
+# Looking for significance with non-missing income and wealth
+non_missing <- test %>%
+    filter(!is.na(pincome),
+           !is.na(pnetworth))
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_w + race_b + mom_educ_hs + dad_educ_hs + ret_sav + hown + both_pars + par_dec, data = non_missing))
+#Non-significant: race_w and par_dec - dropping for now
+
+# Checking for multi-collinearity
+testing_cor <- test %>%
+    select(-id, -hispanic, -race, -mom_education, -dad_education, -savings, -home, -both_parents, -par1_deceased, -par2_deceased, - mom_age_birth, -pincome, -pnetworth, -race_w, -par_dec) %>%
+    drop_na()
+
+M <- cor(testing_cor)
+corrplot(M, type = "upper")
+
+summary(model <- lm(pincome ~ mom_age, data = test)) #Adj R2 = 0.057
+summary(model <- lm(pincome ~ mom_age + hisp, data = test)) #Adj R2 = 0.09
+car::vif(model)
+summary(model <- lm(pincome ~ mom_age + hisp + race_b, data = test)) # #Adj R2 = 0.154
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs, data = test)) #Adj R2 = 0.1767
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs, data = test)) #Adj R2 = 0.1336
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs + ret_sav, data = test)) #Adj R2 = 0.2047 - biggest R2 jump
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs + ret_sav + hown, data = test)) #Adj R2 = 0.2165
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs + ret_sav + hown + both_pars, data = test)) #Adj R2 = 0.2178
+car::vif(model)
+
+# Making df for imputation
+impute <- test %>%
+    select(-hispanic, -race, -mom_education, -dad_education, -savings, -home, -both_parents, -par1_deceased, -par2_deceased, - mom_age_birth, -race_w, -par_dec, -age_resp)
+t <- nearZeroVar(impute) # no variables are found to have near zero variance
+
+# Checking for missings in the predictor variables
+sapply(impute, function(x) sum(is.na(x)))
+
+# Paper that uses multiple imputation: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6049093/
+# Multiple imputation in R reference: https://library.virginia.edu/data/articles/getting-started-with-multiple-imputation-in-r
+init = mice(impute, maxit=10)
+meth = init$method
+predM = init$predictorMatrix
+
+# Set distributions
+num <- c("mom_age") # numerical variables
+log <- c("hisp", "race_b", "mom_educ_hs", "dad_educ_hs", "ret_sav", "hown", "both_pars") # binary variables
+
+meth[num] = "norm" # numerical variables
+meth[log] = "logreg" # binary variables
```

Comment:
I don't think this is doing anything right? Since you're setting them all to "" in line 101 below. And looks like we're using the default "pmm" option for the pincome and pnetworth

Reply:
True - this was part of an older attempt - deleted.